vllm/engine/protocol.py

```diff
@@ -69,24 +70,50 @@ async def beam_search(
         ignore_eos = params.ignore_eos
         temperature = params.temperature
         length_penalty = params.length_penalty
+        include_stop_str_in_output = params.include_stop_str_in_output
 
         tokenizer = await self.get_tokenizer(lora_request=None)
-        tokenizedPrompt = prompt if isinstance(
-            prompt, list) else tokenizer.encode(prompt)
-        tokenizedLength = len(tokenizedPrompt)
+
+        if isinstance(prompt, dict):
+            if "prompt" in prompt:
+                tokenized_prompt = tokenizer.encode(prompt.get("prompt"))
+                multi_modal_data = prompt.get("multi_modal_data")
+                mm_processor_kwargs = prompt.get("mm_processor_kwargs")
+            elif "prompt_token_ids" in prompt:
+                tokenized_prompt = prompt.get("prompt_token_ids")
+                multi_modal_data = prompt.get("multi_modal_data")
+                mm_processor_kwargs = prompt.get("mm_processor_kwargs")
+            else:
+                raise TypeError(
+                    "Dictionary input must be a TextPrompt or TokensPrompt")
+        else:
+            tokenized_prompt = prompt if isinstance(
+                prompt, list) else tokenizer.encode(prompt)
+            multi_modal_data = None
+            mm_processor_kwargs = None
+
+        tokenized_length = len(tokenized_prompt)
 
         sort_beams_key = create_sort_beams_key_function(
             tokenizer.eos_token_id, length_penalty)
 
-        beam_search_params = SamplingParams(logprobs=2 * beam_width,
-                                            max_tokens=1,
-                                            temperature=temperature)
-        all_beams = [BeamSearchSequence(tokens=tokenizedPrompt, cum_logprob=0)]
+        beam_search_params = SamplingParams(
+            logprobs=2 * beam_width,
+            max_tokens=1,
+            temperature=temperature,
+        )
+        all_beams = [
+            BeamSearchSequence(tokens=tokenized_prompt, cum_logprob=0)
```

Comment:
Similar to the `Sequence` we use normally, let's place `multi_modal_data` and `mm_processor_kwargs` inside `BeamSearchSequence`.

Reply:
Thank you for the comment, I added `multi_modal_data` and `mm_processor_kwargs` to the class